lib/bambora/v1/profile_resource.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+module Bambora
+  module V1
+    class ProfileResource
+      def initialize(client:, sub_path:)
+        @client = client
+        @sub_path = sub_path
```

Comment:
Simply pass in the client and the sub-path. The reason the sub-path is needed is because when instantiating an instance of `Excon`, whatever is passed in as the base URL will get cut off after the last domain (e.g. `.com`).

Therefore,

```ruby
Excon.new('https://bambora.com/v1/profiles')
```

and

```ruby
Excon.('https://bambora.com')
```

are equivalent. All requests will be made to `https://bambora.com`, and the remaining path needs to be passed in when making a request against the instance.

Reply:
While this is an implementation short-coming of `Excon`, I don't see why the developer ergonomics require passing in the `sub_path`. Wouldn't it always be consistent for all profiles, and therefore be an implementation detail of the class, rather than developer knowledge passed in?